packages/widgets/src/Transfer/OptionsContainer.js

```diff
@@ -0,0 +1,144 @@
+import { CircularLoader } from '@dhis2/ui-core'
+import { spacers } from '@dhis2/ui-constants'
+import React, { Fragment, useEffect, useRef, useState } from 'react'
+import propTypes from '@dhis2/prop-types'
+
+import { EndIntersectionDetector } from './EndIntersectionDetector.js'
+
+export const OptionsContainer = ({
```

Comment:
I know we talked about the IntersectionDetector not working properly in some cases. I suspect that you've addressed that issue here with the resize observer? Or is it still an issue?

I think what I'd prefer as a reviewer to fix the intersection detector bug is a separate PR with a failing test as the initial commit. So that we can then test our solutions against that. From the reviewer's (or my) perspective that'd be a nice approach.

One other thing I'm still skeptical about with the IntersectionDetector component by the way is that the name is directly tied to the api. The way I see it, we're interested in detecting visibility. That's why I preferred VisibilityDetector in the other PR for this component, since it seemed like that was what it had to do.

Now instead of the function we've tied the naming to the api. And now that we're encountering issues where the api isn't fulfilling our needs of detecting visibility we can't fix it within the component, because it doesn't fit the scope of it. I think I'd still prefer a visibility detector component that neatly packages that functionality. That way we can just use that component to detect when the end of the list has been reached, and fix bugs within that component.

Reply:
Yes, I've addressed it already. The problem with the visibility component in this case is, that the fix had to be done for a DOM element that's specific to the transfer.

I'd prefer to stay close to the actual api the component is implementing. The component is actually testing an intersection. The issue that came up wasn't an issue of the intersection detector alone, but it interacting with the resize observer.

I think the only possible solution I can think of is a `VisibleContainer` and a `VisibilityDetector` which have to be used in combination, as the setup will always be something like this:


```
container
  > some custom content
    > detector
```

The issue is that the `detector` needs to be a child of the custom content, which would increase the complexity of the component. I prefer to stick to the actual functionality of the component as it's clear what you can use it for. If we ever want to have some kind of visibility detector component, it should be separate of course